tests/functions_tests/test_mean_squared_error.py

```diff
@@ -0,0 +1,148 @@
+import numpy
+import pytest
+
+import chainer
+from chainer import cuda
+from chainer import gradient_check
+
+import chainer_chemistry
+
+
+@pytest.fixture
+def inputs():
+    x0 = numpy.random.uniform(-1, 1, (4, 3)).astype(numpy.float32)
+    x1 = numpy.random.uniform(-1, 1, (4, 3)).astype(numpy.float32)
+    x2 = numpy.asarray([[0.3, numpy.nan, 0.2],
+                        [numpy.nan, 0.1, 0.5],
+                        [0.9, 0.7, numpy.nan],
+                        [0.2, -0.3, 0.4]]).astype(numpy.float32)
+    return x0, x1, x2
+
+
+@pytest.fixture
+def grads():
+    gy = numpy.random.uniform(-1, 1, ()).astype(numpy.float32)
+    ggx0 = numpy.random.uniform(-1, 1, (4, 3)).astype(numpy.float32)
+    ggx1 = numpy.random.uniform(-1, 1, (4, 3)).astype(numpy.float32)
+    return gy, ggx0, ggx1
+
+
+def check_forward(inputs):
+    x0_data, x1_data, _ = inputs
+    x0 = chainer.Variable(x0_data)
+    x1 = chainer.Variable(x1_data)
+    loss = chainer_chemistry.functions.mean_squared_error(x0, x1)
+    loss_value = cuda.to_cpu(loss.data)
+    assert loss.dtype == numpy.float32
+    assert loss_value.shape == ()
+
+    loss_expect = 0.
+    for i in numpy.ndindex(x0_data.shape):
+        loss_expect += (x0_data[i] - x1_data[i]) ** 2
+    loss_expect /= x0_data.size
+    assert pytest.approx(loss_value, loss_expect)
+
+
+def check_forward_ignore_nan(inputs):
+    x0_data, _, x2_data = inputs
+    x0 = chainer.Variable(x0_data)
+    x2 = chainer.Variable(x2_data)
+    loss = chainer_chemistry.functions.mean_squared_error(x0, x2,
+                                                          ignore_nan=True)
+    loss_value = cuda.to_cpu(loss.data)
+    assert loss.dtype == numpy.float32
+    assert loss_value.shape == ()
+
+    loss_expect = 0.
+    x0_data = cuda.to_cpu(x0_data)
+    x2_data = cuda.to_cpu(x2_data)
+    nan_mask = numpy.invert(numpy.isnan(x2_data)).astype(x2_data.dtype)
+    for i in numpy.ndindex(x0_data.shape):
+        loss_expect += ((x0_data[i] - numpy.nan_to_num(x2_data[i])) ** 2
+                        * nan_mask[i])
+    loss_expect /= x0_data.size
+    assert numpy.allclose(loss_value, loss_expect)
```

Comment:
Why did you use `numpy.allclose` in this function, while `pytest.approx` in `check_forward`?

Reply:
Fix this. Thank you.